### PR TITLE
+ added body prepend and append slots

### PR DESCRIPTION
**Pull request by NovalisTheMagician**
### `Type`

> What types of changes does your code introduce?

Added prepend and append slots to the table body similar to how [vuetifiyjs v2](https://vuetifyjs.com/en/api/v-data-table/#slots-body.prepend) is doing it

These slots represent the first and last row of the visible table and are always present (even if no data is available).
That space might be used to implement a per column search or whole column copies.

> Put an `x` in the boxes that apply_

- [ ] Fix
- [x] Feature

### `Checklist`

> Put an `x` in the boxes that apply._

- [x] Title as described
- [ ] Add unit test by vitest if necessary

### CODE REVIEW

**Comment by HC200ok** [comment]
@NovalisTheMagician Thank you for the PR.

You added `prepend and append slots` but you didn't bind any variables or methods like this:
```vue
<slot
  name="pagination"
  v-bind="{
    isFirstPage,
    isLastPage,
    currentPaginationNumber,
    maxPaginationNumber,
    nextPage,
    prevPage,
  }"
/>
```

Please comment with the description of the feature (for example: show some headers related information at the start of the default table <tbody>)of `prepend and append` slots. Don't forget to bind the related variables and methods for achieving the feature.

And it is better to give an example of how you use the `prepend and append` slots in the [client mode](https://github.com/HC200ok/vue3-easy-data-table/blob/main/src/modes/Client.vue) or [server-side mode](https://github.com/HC200ok/vue3-easy-data-table/blob/main/src/modes/ServerSide.vue)

**Comment by NovalisTheMagician** [comment]
I've added the bound variables and updated the client mode example with usage of the slots

File: src/components/DataTable.vue [hunk 1]
```diff
@@ -76,10 +76,25 @@
           </tr>
         </thead>
         <tbody
-          v-if="items.length && headerColumns.length"
```
Comment (HC200ok) [inline src/components/DataTable.vue:79]: If `items` is empty,
`items` and  `pagination` binded in
```
<slot 
  name="body.prepend"
  v-bind="{
    items: pageItems,
    pagination: {
        isFirstPage,
        isLastPage,
        currentPaginationNumber,
        maxPaginationNumber,
        nextPage,
        prevPage
     },
     headers: headersForRender
   }"
/>
```

will have no meaning

File: src/components/DataTable.vue [hunk 2]
```diff
@@ -140,6 +155,21 @@
               </td>
             </tr>
           </template>
+          <slot 
+            name="body.append"
```
Comment (HC200ok) [inline src/components/DataTable.vue:159]: For the uniform style, '-' is recommended instead of '.'

**Pull request merged**